applications: serial_lte_modem: Store credentials to settings (#13610)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/slm_description.rst

.. _CONFIG_SLM_NATIVE_TLS:

CONFIG_SLM_NATIVE_TLS - Use Zephyr's Mbed TLS for TLS connections
   This option is enabled by the Native TLS overlay.
   See :ref:`slm_native_tls` for more information.

> CONFIG_SLM_NATIVE_TLS_CREDENTIAL_BUFFER_SIZE - Buffer space reserved for loading credentials

Reviewer: ```suggestion
.. _CONFIG_SLM_NATIVE_TLS_CREDENTIAL_BUFFER_SIZE:

CONFIG_SLM_NATIVE_TLS_CREDENTIAL_BUFFER_SIZE - Buffer space reserved for loading credentials
```
Configs need to have the target tag for linking.

>    Specifies the credential buffer size available for a single sec_tag when loading credentials for Mbed TLS.

Reviewer: ```suggestion
   Specifies the credential buffer size available for a single ``sec_tag`` when loading credentials for Mbed TLS.
```
Or should it be `<sec_tag>` like it is given in the AT command documentation?

Author: Likely it should be ``sec_tag`` when not used as a parameter.

>    The default value is 4096.

Reviewer: ```suggestion
   The default value is ``4096``.
```
Since it's a value.

> CONFIG_SLM_NATIVE_TLS_CREDENTIAL_BUFFER_COUNT - Amount of buffers for loading credentials

Reviewer: ```suggestion
.. _CONFIG_SLM_NATIVE_TLS_CREDENTIAL_BUFFER_COUNT:

CONFIG_SLM_NATIVE_TLS_CREDENTIAL_BUFFER_COUNT - Number of buffers for loading credentials
```

>    Amount of buffers available for loading sec_tag credentials for Mbed TLS.

Reviewer: ```suggestion
   The number of buffers available for loading ``sec_tag`` credentials for Mbed TLS.
```
Question on formatting applies here as well.

>    TLS Client needs the buffer only when connecting where as TLS server needs the buffer as long as it running.

Reviewer: ```suggestion
   TLS client only needs the buffer when connecting, while TLS server needs the buffer as long as it running.
```

>    Increase the value, if you need TLS client and server running simultaneously with different sec_tags.

Reviewer: ```suggestion
   Increase the value if you need both TLS client and server running simultaneously with different ``sec_tags``.
```

>    The default value is 1.

Reviewer: ```suggestion
   The default value is ``1``.
```

.. _CONFIG_SLM_NATIVE_TLS:

CONFIG_SLM_NATIVE_TLS - Use Zephyr's Mbed TLS for TLS connections

>    This option is enabled by the Native TLS overlay.

Reviewer 2: ```suggestion
   This option is enabled by the native TLS overlay.
```

Native TLS
----------

By default, the secure socket (TLS) is offloaded onto the modem.
If you need customized TLS features that are not supported by the modem firmware, you can use native TLS instead.
Native TLS uses the Mbed TLS library in Zephyr to establish secure connectivity.
Currently, the SLM application can be built to use native TLS for the following services:

* Secure socket

> * TLS Proxy client/server

Reviewer: ```suggestion
* TLS Proxy client
* TLS Proxy server
```

Native TLS
----------

> By default, the secure socket (TLS) is offloaded onto the modem.

Reviewer 2: ```suggestion
By default, the secure socket (TLS) is offloaded to the modem.
```

If you need customized TLS features that are not supported by the modem firmware, you can use native TLS instead.
Native TLS uses the Mbed TLS library in Zephyr to establish secure connectivity.
Currently, the SLM application can be built to use native TLS for the following services:

* Secure socket
* TLS Proxy client
* TLS Proxy server
* HTTPS client

With native TLS, the credentials are stored in the Zephyr settings storage with the ``AT#XCMNG`` command.

> The configuration options that are required to enable the native TLS socket are defined in the :file:`overlay-native_tls.conf` file.

Reviewer 2: ```suggestion
The configuration options that are required to enable native TLS are defined in the :file:`overlay-native_tls.conf` file.
```

.. note::

   Native TLS services have the following limitations:

>    * PSK, PSK identity, and PSK public key are currently not supported.

Reviewer 2: Though they (PSK, PSK identity) are defined in the AT command and the code. Is this planned to be fixed/supported?

Author: Good catch. It is supported now.

Reviewer 2: This still needs to be fixed.

Reviewer 2: What about the type `5 – Public key (ASCII text)`? It is not defined in the header nor the documentation, so it seems like it should be added here.

Author: It is for completely different use case than native TLS. I think that it has been mistake to have it documentation to begin with.
 – Public key (ASCII text). Used in authenticated AT commands.
